drivers/potentiometer/common/dpot.h

```diff
@@ -0,0 +1,308 @@
+ */
+enum dpot_dev_id {
+	DEV_AD5121,
+	DEV_AD5122,
+	DEV_AD5122A,
+	DEV_AD5123,
+	DEV_AD5124,
+	DEV_AD5141,
+	DEV_AD5142,
+	DEV_AD5142A,
+	DEV_AD5143,
+	DEV_AD5144,
+	DEV_AD5160,
+	DEV_AD5161,
+	DEV_AD5165,
+	DEV_AD5171,
+	DEV_AD5241,
+	DEV_AD5242,
+	DEV_AD5245,
+	DEV_AD5246,
+	DEV_AD5258,
+	DEV_AD5259,
+	DEV_AD5273
+};
+#define NUM_SUPPORTED_DEVICES DEV_AD5273+1
+
+/**
+ * @enum dpot_intf_type.
+ * @brief digital potentiometer interface type.
+ */
+enum dpot_intf_type {
+	/* SPI digital interface */
+	AD_SPI_INTERFACE = 0x1,
+	/* I2C digital interface */
+	AD_I2C_INTERFACE = 0x2
+};
+/**
+ * @enum dpot_chn_type.
+ * @brief Digital potentiometer channel type.
+ */
+enum dpot_chn_type {
+	/* Channels applicable in potentiometer mode */
+	DPOT_CHN_RDAC1, /* Use this channel for single channel devices */
+	DPOT_CHN_RDAC2,
+	DPOT_CHN_RDAC3,
+	DPOT_CHN_RDAC4,
+	/* Channel applicable in linear gain setting mode */
+	/* Use R_AW1 and R_WB1 for single channel devices */
+	DPOT_CHN_R_AW1,
+	DPOT_CHN_R_WB1,
+	DPOT_CHN_R_AW2,
+	DPOT_CHN_R_WB2,
+	DPOT_CHN_R_AW3,
+	DPOT_CHN_R_WB3,
+	DPOT_CHN_R_AW4,
+	DPOT_CHN_R_WB4,
+	NUM_OF_DPOT_CHN
+};
+
+
+/**
+ * @enum dpot_operating_mode.
+ * @brief Digital potentiometer operating mode.
+ * @note Operating mode is applicable to only certain
+ *       generics of digipots and not all.
+ */
+enum dpot_operating_mode {
+	/* Potentiometer mode */
+	DPOT_POTENTIOMETER_MODE,
+	/* Linear gain setting mode */
+	DPOT_LINEAR_GAIN_SETTING_MODE
+};
+
+/**
+ * @enum dpot_rdac_linear_status.
+ * @brief Digital potentiometer RDAC linear update status.
+ * @note Linear update is applicable to only certain
+ *       generics of digipots and not all.
+ */
+enum dpot_rdac_linear_status {
+	/* Linear increment */
+	DPOT_RDAC_LINEAR_INCREMENT,
+	/* Linear decrement */
+	DPOT_RDAC_LINEAR_DECREMENT
+};
+
+/**
+ * @enum dpot_rdac_6db_status.
+ * @brief Digital potentiometer RDAC 6dB update status.
+ * @note 6dB update is applicable to only certain
+ *       generics of digipots and not all.
+ */
+enum dpot_rdac_6db_status {
+	/* 6dB increment */
+	DPOT_RDAC_6DB_INCREMENT,
+	/* 6dB decrement */
+	DPOT_RDAC_6DB_DECREMENT
+};
+
+/**
+ * @struct dpot_command.
+ * @brief Digital potentiometer command structure.
+ */
+struct dpot_command {
+	/* Control bits (DB15:DB12) */
+	uint8_t control;
+	/* Address bits (DB11:DB8) */
+	uint8_t address;
+	/* Data bits (DB7:DB0) */
+	uint8_t data;
+	/* Readback needed (optional response to command) */
+	bool is_readback;
+	/* Response (optional, if 'is_readback' is set true) */
+	uint8_t response;
+};
+/**
+ * @struct dpot_init_param.
+ * @brief Digital potentiometer init parameters.
+ */
+typedef struct dpot_init_param {
+
+	enum dpot_intf_type intf_type;
+	/* Active device ID */
+	enum dpot_dev_id device_id;
+	/* Pointer to digipots ops (APIs) structure */
+	const struct dpot_ops *dpot_ops;
+	/* Device specific parameters */
+	void *extra;
+	enum dpot_operating_mode operating_mode;
+} DPOT_INIT_PARAM;
```

Comment:
as per our coding guideline, we dont use typedefs, and especially not in capslock... 

i suggest to remove the typedef keyword and name and use `struct dpot_init_param` everywhere